src/js/demo/components/App.jsx

```diff
@@ -147,9 +161,11 @@ export default class App extends Component {
     }
 
     render() {
-        const results = this.lint();
+        const { results: { messages, output }, error } = this.lint();
         const sourceCode = linter.getSourceCode();
         const { text, fix, options } = this.state;
+        const isInvalidAutofix = !error && fix && text !== output &&
```

Comment:
I'm a bit concerned by the complexity of this check. Is there a way it could be decomposed to improve readability? Or, could we change the returned object (from `lint()`) to make the result a bit more clear and simplify this check?

Reply:
Changed the returned object from `lint()`. The logic should be more readable now, but I'll change it again if it could be better.